docs/solutions/sql_server_always_on_availability_groups/modules/iam/compartment/main.tf

```diff
@@ -0,0 +1,6 @@
+resource "oci_identity_compartment" "compartment" {
+  count       = "${var.compartment_id == "" ? 1 : 0}"
```

Comment:
This is a pretty neat work around, but I don't believe it's necessary. The compartment resource will actually look for a compartment with the specified name, and if it already exists it will just use that one instead of creating a new one. So I would specify only name and description here, and once you do that there's probably not a good reason to make this a module either.

Reply:
It actually is necessary due to to a bug in Terraform. Not putting this causes it to fail :)

Reply:
I haven't tried with this config, but testing a similar approach in another config gives me the error "value of 'count' cannot be computed", due to issue https://github.com/hashicorp/terraform/issues/10857. 

If there resource already exists I don't see the error, but if I destroy and apply again, then I see it. In this case you would want to test where compartment ID is and is not set on a fresh apply.

Reply:
This comment still needs to be addressed. I would be in favor or removing count and var.compartment_id. Actually, I would completely get rid of the compartment module - just use the resource directly.